### PR TITLE
Add Dataset::build_overviews

### PR DESCRIPTION
**Pull request by lnicola**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/gdal/blob/master/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

### CODE REVIEW

File: src/dataset.rs [hunk 1]
```diff
@@ -315,6 +317,39 @@ impl Dataset {
...
+    /// Builds overviews for the current `Dataset`. See [`GDALBuildOverviews`].
+    ///
+    /// # Arguments
+    /// * `resampling` - resampling method, as accepted by GDAL, e.g. `"CUBIC"`
+    /// * `overviews` - list of overview decimation factors, e.g. `&[2, 4, 8, 16, 32]`
+    /// * `bands` - list of bands to build the overviews for, or empty for all bands
+    ///
+    /// [`GDALBuildOverviews`]: https://gdal.org/doxygen/gdal_8h.html#a767f4456a6249594ee18ea53f68b7e80
+    pub fn build_overviews(
+        &self,
```
Comment (michaelkirk) [inline src/dataset.rs:329]: In line with https://github.com/georust/gdal/pull/161, I wonder if @rmanoka or @jdroenner have thoughts about whether this should be &mut.

Comment (frewsxcv) [inline src/dataset.rs:329]: Should this take `&mut self` if it's modifying `self.c_dataset`?

**Review by frewsxcv — APPROVED** [review]
Besides my comment, lgtm!

File: src/dataset.rs (hunk 1 above)
Comment (frewsxcv) [reply src/dataset.rs:329]: Lol @michaelkirk just commented about the same thing :)

**Comment by lnicola** [comment]
Updated.

Unrelated, tests fail intermittently on my computer:

  process didn't exit successfully: `/home/grayshade/Projects/georust/gdal/target/debug/deps/gdal-4103840a8afd223b` (signal: 11, SIGSEGV: invalid memory reference)

File: src/dataset.rs (hunk 1 above)
Comment (jdroenner) [reply src/dataset.rs]: i guess &mut is the better choice.

**Pull request merged**